### ISSUE TITLE
Accessibility violation in Datagrid with 'Infinite scroll' and 'Clickable Row'

### ISSUE THREAD

**Issue by laukf** [carbon-design-system/ibm-products#3000]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

IBM Equal Access Accessibility Checker reports two violations on a table created using Datagrid infinite scroll and clickable row:
<img width="530" alt="Screenshot 2023-05-15 at 16 50 23" src="https://github.com/carbon-design-system/ibm-cloud-cognitive/assets/133672377/137b512a-082f-44d0-9418-751768f1438b">

For the first violation, affected element:
<img width="341" alt="Screenshot 2023-05-15 at 16 50 55" src="https://github.com/carbon-design-system/ibm-cloud-cognitive/assets/133672377/8ddefa9b-207b-40dd-bbbd-e64174b2f9a8">

For the second violation, affected element:
<img width="742" alt="Screenshot 2023-05-15 at 16 51 47" src="https://github.com/carbon-design-system/ibm-cloud-cognitive/assets/133672377/14da09b1-f098-467f-b8a7-3898bc9aa01f">

> Is this issue related to a specific component?

Yes, Datagrid with infinite scroll and clickable row 

> What did you expect to happen? 

Absence of accessibility violation.

> What browser are you working in?

Firefox, Chrome

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

v1.45.0

## Steps to reproduce the issue
1. Create a table using Datagrid and infinite scrolling and clickable row
2. Run IBM Equal Access Accessibility Checker on this table

**Comment by elycheea** [comment]
@laukf Is this still a current issue? Please provide a CodeSandbox or Stackblitz or at least minimum steps to reproduce this issue.
